backend/internal/api/server.go

```diff
@@ -11,25 +13,31 @@ import (
 
 // APIServer represents an HTTP server for the JSON API.
 type APIServer struct {
-	address       string
-	userDB        entity.UserDB
-	associationDB entity.AssociationDB
-	invitationDB  entity.InvitationDB
-	bucketDb      entity.BucketDB
-	userService   *service.UserService
+	address            string
+	bucketDb           entity.BucketDB
+	userService        *service.UserService
+	associationService *service.AssociationService
+	invitationService  *service.InvitationService
 }
 
 // NewAPIServer creates a new instance of APIServer with the given address.
 func NewAPIServer(
 	address string,
-	bucketDb entity.BucketDB,
-	userDB entity.UserDB,
 ) *APIServer {
+	ctx := context.Background()
```

Comment:
why are we moving this stuff to here? I think they should be kept in the main file